File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.java
class Solution {
    public int characterReplacement(String s, int k) {
        int arr[] = new int[26];
        int i = 0;
        int j = 0;
        int max = 0;
        int res = 0;
        while(j<s.length()){
            arr[s.charAt(j) - 'A']++;
            max = Math.max(max, arr[s.charAt(j) - 'A']);
            if(j-i+1 - max > k){
                arr[s.charAt(i) - 'A']--;
                i++;
            }
            res = Math.max(res, j-i+1);
            j++;
        }
        return res;
    }
}

File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.cs
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

public class Solution {
    public int MinMeetingRooms(List<Interval> intervals) {
        int[] start = new int[intervals.Count];
        int[] end = new int[intervals.Count];
        for(int i=0;i<intervals.Count;i++){
            start[i] = intervals[i].start;
            end[i] = intervals[i].end;
        }
        Array.Sort(start);
        Array.Sort(end);
        int res = 0;
        int count = 0;
        int s1 = 0, e1 = 0;
        while(s1<intervals.Count){
            if(start[s1]<end[e1]){
                count++;
                s1++;
            }
            else{
                count--;
                e1++;
            }
            res = Math.Max(res, count);
        }
        return res;
    }
}
